File: 3046-split-the-array/3046-split-the-array.cpp
class Solution {
public:
    bool isPossibleToSplit(vector<int>& nums) {
        int n = nums.size();
        if(n % 2 != 0) return false;
        map<int,int> m;
        for(int num : nums) m[num]++;
        for(int num : nums) {
            if(m[num] > 2) return false;
        }
        return true;

    }
};

File: 3046-split-the-array/3046-split-the-array.java
class Solution {
    public boolean isPossibleToSplit(int[] nums) {
        int n = nums.length;
        if(n % 2 != 0) return false;
        HashMap<Integer,Integer> map = new HashMap<>();
        for(int num : nums) {
             map.put(num,map.getOrDefault(num,0)+1);
        }
        for(int num : nums) {
             if(map.get(num) > 2) return false;
        }
        return true;
    }
}